sdc/datatypes/hpat_pandas_series_functions.py

```diff
@@ -105,77 +244,212 @@ def hpat_pandas_series_getitem(self, idx):
     _func_name = 'Operator getitem().'
 
     if not isinstance(self, SeriesType):
-        raise TypingError('{} The object must be a pandas.series. Given: {}'.format(_func_name, self))
+        return None
 
-    if isinstance(idx, types.Integer):
-        def hpat_pandas_series_getitem_idx_integer_impl(self, idx):
-            """
-            **Test**: python -m sdc.runtests sdc.tests.test_series.TestSeries.test_series_iloc1
-            """
+    # Note: Getitem return Series
+    index_is_none = self.index is None or isinstance(self.index, numba.types.misc.NoneType)
+    index_is_number = index_is_none or (self.index and isinstance(self.index.dtype, types.Number))
+    index_is_string = not index_is_none and isinstance(self.index.dtype, (types.UnicodeType, types.StringLiteral))
+
+    if (
+        isinstance(idx, types.Number) and index_is_number or
+        (isinstance(idx, (str, types.UnicodeType, types.StringLiteral)) and index_is_string)
```

Comment:
There's no need to check for str here, as idx has no default value, and hence it should never have pure Python type, i.e. (types.UnicodeType, types.StringLiteral)